games/src/main/java/brocoeur/example/games/service/roulette/RouletteService.java

```diff
@@ -1,83 +1,86 @@
-        return new PlayerResponse(
-                123,
-                Integer.parseInt(player.getUserId()),
-                isWinner,
-                player.getAmountToGamble(),
-                player.getLinkedJobId());
-    }
+        int availableMoney = playerRequest.getAmountToGamble();
+        final RouletteGameStrategy rouletteGameStrategy = (RouletteGameStrategy) playerRequest.getGameStrategyTypes().getGameStrategy();
 
-    public PlayerResponse play(final PlayerRequest player, final List<Boolean> listOfIsWinner) {
-        final GamePlay userPlay = player.getOfflineGameStrategyTypes().getOfflineGameStrategy().playOffline(listOfIsWinner);
-        final GamePlay servicePlay = play();
-        final boolean isWinner = servicePlay.equals(userPlay);
-        return new PlayerResponse(
-                123,
-                Integer.parseInt(player.getUserId()),
-                isWinner,
-                player.getAmountToGamble(),
-                player.getLinkedJobId());
-    }
+        for (var i = 0; i < ttl; i++) {
+            final List<Gamble> gambleList = rouletteGameStrategy.play(availableMoney, previousRoulettePlay);
+            availableMoney -= gambleList.stream().mapToInt(Gamble::amount).sum();
 
-    /**
-     * Use for <b>Direct</b> play.
-     */
-    public List<PlayerResponse> playRouletteGame(final ServiceRequest serviceRequest) {
-        LOGGER.info("Game of roulette started for : {}", serviceRequest);
+            final RoulettePlay servicePlay = playServiceRoulette();
+            previousRoulettePlay.add(servicePlay);
 
-        final PlayerRequest playerRequest = serviceRequest.getPlayerRequestList().get(0);
-        final PlayerResponse playerResponse = play(playerRequest);
+            final int amountWon = getAmountWon(gambleList, servicePlay);
+            availableMoney += amountWon;
+        }
+        final PlayerResponse playerResponse = new PlayerResponse(
+                123,
+                Integer.parseInt(playerRequest.getUserId()),
+                playerRequest.getAmountToGamble(),
+                availableMoney,
+                playerRequest.getLinkedJobId()
+        );
 
-        LOGGER.info("Game of roulette ended with response : {}", playerResponse);
+        LOGGER.info("Game of roulette ended with response: {}", playerResponse);
 
         return List.of(playerResponse);
     }
 
-    /**
-     * Use for <b>Offline</b> play.
-     */
-    public List<PlayerResponse> playRouletteGame(final ServiceRequest serviceRequest, final List<Boolean> listOfIsWinner) {
-        LOGGER.info("Game of roulette started for : {}", serviceRequest);
-
-        final PlayerRequest playerRequest = serviceRequest.getPlayerRequestList().get(0);
-        final PlayerResponse playerResponse = play(playerRequest, listOfIsWinner);
-
-        LOGGER.info("Game of roulette ended with response : {}", playerResponse);
+    private int getAmountWon(final List<Gamble> gambleList, final RoulettePlay servicePlay) {
```

Comment:
Same as before : no money has been yet, if the player looses this name of method isn't right

Reply:
next mr.